typings/__tests__/index.test.tsx

```diff
@@ -161,105 +165,50 @@ const findAllBy: Promise<ReactTestInstance[]>[] = [
-  'button'
-);
-
-const getByA11yStates: ReactTestInstance = tree.getByA11yStates('selected');
-const getByA11yStatesArray: ReactTestInstance = tree.getByA11yStates([
-  'selected',
-]);
-const getAllByA11yStates: Array<ReactTestInstance> = tree.getAllByA11yStates(
-  'selected'
-);
-const getAllByA11yStatesArray: Array<ReactTestInstance> = tree.getAllByA11yStates(
-  ['selected']
-);
-const queryByA11yStates: ReactTestInstance = tree.queryByA11yStates('selected');
-const queryByA11yStatesArray: ReactTestInstance = tree.queryByA11yStates([
-  'selected',
-]);
-const queryAllByA11yStates: Array<ReactTestInstance> = tree.queryAllByA11yStates(
-  'selected'
-);
-const queryAllByA11yStatesArray: Array<ReactTestInstance> = tree.queryAllByA11yStates(
-  ['selected']
-);
-
-const getByA11yState: ReactTestInstance = tree.getByA11yState({ busy: true });
-const getAllByA11yState: Array<ReactTestInstance> = tree.getAllByA11yState({
-  busy: true,
-});
-const queryByA11yState: ReactTestInstance = tree.queryByA11yState({
-  busy: true,
-});
-const queryAllByA11yState: Array<ReactTestInstance> = tree.queryAllByA11yState({
-  busy: true,
-});
-
-const getByA11yValue: ReactTestInstance = tree.getByA11yValue({ min: 10 });
-const getAllByA11yValue: Array<ReactTestInstance> = tree.getAllByA11yValue({
-  min: 10,
-});
-const queryByA11yValue: ReactTestInstance = tree.queryByA11yValue({ min: 10 });
-const queryAllByA11yValue: Array<ReactTestInstance> = tree.queryAllByA11yValue({
-  min: 10,
-});
-
+// debug API
 const debugFn = tree.debug();
 const debugFnWithMessage = tree.debug('my message');
 
+// update API
 tree.update(<View />);
 tree.rerender(<View />);
 tree.unmount();
 
-// fireEvent API tests
-fireEvent(getByA11yLabel, 'press');
-fireEvent(getByA11yLabel, 'press', 'data');
-fireEvent(getByA11yLabel, 'press', 'param1', 'param2');
-fireEvent.press(getByA11yLabel);
-fireEvent.changeText(getByA11yLabel, 'string');
-fireEvent.scroll(getByA11yLabel, 'eventData');
+// fireEvent API
+const element: ReactTestInstance = tree.getByText('text');
+fireEvent(element, 'press');
+fireEvent(element, 'press', 'data');
+fireEvent(element, 'press', 'param1', 'param2');
+fireEvent.press(element);
+fireEvent.changeText(element, 'string');
+fireEvent.scroll(element, 'eventData');
 
-// shallow API
-const shallowTree: { output: React.ReactElement<any> } = shallow(
-  <TestComponent />
-);
+// waitFor API
+const waitGetBy: Promise<ReactTestInstance>[] = [
+  waitFor<ReactTestInstance>(() => tree.getByA11yLabel('label')),
+  waitFor<ReactTestInstance>(() => tree.getByA11yLabel('label'), {
+    timeout: 10,
+  }),
+  waitFor<ReactTestInstance>(() => tree.getByA11yLabel('label'), {
+    timeout: 100,
+    interval: 10,
+  }),
+];
 
-const waitForFlush: Promise<any> = flushMicrotasksQueue();
+const waitGetAllBy: Promise<ReactTestInstance[]>[] = [
+  waitFor<ReactTestInstance[]>(() => tree.getAllByA11yLabel('label')),
```

Comment:
`waitFor` can work with any throwable function, maybe we can showcase that and get rid of `tree.getAllBy` examples?